lib/bootloader_setup.pm

```diff
@@ -871,33 +871,35 @@ sub tianocore_http_boot {
 
 sub zkvm_add_disk {
     my ($svirt) = @_;
-    if (my $hdd = get_var('HDD_1')) {
-        my $basename = basename($hdd);
-        my $basedir  = svirt_host_basedir();
-        my $hdd_dir  = "$basedir/openqa/share/factory/hdd";
-        my $hdd_path = $svirt->get_cmd_output("find $hdd_dir -name $basename | head -n1 | tr -d '\n'");
-        die "Unable to find image $basename in $hdd_dir" unless $hdd_path;
-        diag("HDD path found: $hdd_path");
-        if (get_var('PATCHED_SYSTEM')) {
-            diag('in patched systems just load the patched image');
-            my $name        = $svirt->name;
-            my $patched_img = "$zkvm_img_path/$name" . "a.img";
-            $svirt->add_disk({file => $patched_img, dev_id => 'a'});
+    my $dev_id   = 'a';
+    
+    foreach my $n (1 .. get_var('NUMDISKS')) {
```

Comment:
I guess we need to either put default value or check if variable is defined as it might be `undef`.

Reply:
true - thanks for catching

Reply:
fixed